src/main/java/com/aphyr/riemann/client/Promise.java

```diff
@@ -20,14 +21,37 @@ public static Object rehydrate(final Object value) throws IOException {
     }
   }
 
+  public static Object asResult(Object value) {
+    if (value instanceof Exception) {
+      return null;
```

Comment:
Doesn't cause a problem (in the node-style callbacks you always have to check the exception on the caller), but looks like I'll change it to have separate callbacks anyway.